### PR TITLE
Changes long type to string. Fixes issue #20

### PR DESCRIPTION
**Pull request by Zettersten**

### CODE REVIEW

File: IntegrationTests/GistTests.cs [hunk 1]
```diff
@@ -52,7 +52,7 @@ public void Setup()
         [Test]
         async public Task GetGist()
         {
-            const long gistId = 6287413;
+            const string gistId = "6287413";
```
Comment (timerickson) [inline IntegrationTests/GistTests.cs]: let's introduce a non-numeric char here in this value just to show the reason for changing the type

File: IronGitHub/Entities/Gist.cs [hunk 2]
```diff
@@ -123,24 +123,24 @@ public PatchedGistFile(GistFile gistFile)
...
 
         private static readonly Regex GistUrlRegex = new Regex(@"http(?:s)?://(?:api|gist)\.github\.com(?:/gists|/raw|/[^/]+)?/([0-9]+)(?:/.*$)?", RegexOptions.IgnoreCase);
-        public static long ParseIdFromUrl(string url)
+
+        public static string ParseIdFromUrl(string url)
         {
             var match = GistUrlRegex.Match(url);
             if (match.Groups.Count < 2)
-                return -1;
+                return string.Empty;
```
Comment (timerickson) [inline IronGitHub/Entities/Gist.cs]: why not null instead of string.Empty?

File: UnitTests/GistTests.cs [hunk 3]
```diff
@@ -18,7 +18,7 @@ public class GistUrlParseTests
         [Theory]
         public void ParseUrl(string url)
         {
-            Gist.ParseIdFromUrl(url).Should().Be(5731704);
+            Gist.ParseIdFromUrl(url).Should().Be("5731704");
```
Comment (timerickson) [inline UnitTests/GistTests.cs]: add non-numeric char in value like above

**Comment by Zettersten** [comment]
Let me know...

**Comment by Zettersten** [comment]
A lot of older tests are failing. I'm doing some major re-factoring (to the tests that). Hang tight.

**Comment by johnduhart** [comment]
@erik5388 Why are you putting this much effort into a non-offical GitHub client? Why not octokit?

**Comment by Zettersten** [comment]
Kill it...

**Pull request closed without merging**